streamer/controller_node.py

```diff
@@ -73,27 +75,57 @@ def start(self, output_location: str,
             pipeline_config_dict: Dict[str, Any],
             bitrate_config_dict: Dict[Any, Any] = {},
             bucket_url: Union[str, None] = None,
-            check_deps: bool = True) -> 'ControllerNode':
+            check_deps: bool = True,
+            use_hermetic: bool = True) -> 'ControllerNode':
     """Create and start all other nodes.
 
     :raises: `RuntimeError` if the controller has already started.
     :raises: :class:`streamer.configuration.ConfigError` if the configuration is
              invalid.
     """
 
+    if use_hermetic:
+      try:
+        import streamer_binaries # type: ignore
+      except (ImportError, PermissionError) as ex:
+        # If the package couldn't be imported.
+        if isinstance(ex, ImportError):
+          raise RuntimeError(
+              'shaka-streamer-binaries was not found.\n'
+              '  Install it with `pip install shaka-streamer-binaries`.\n'
+              '  Alternatively, use the `--use-system-binaries` option if you '
+              'want to use the system wide binaries of ffmpeg/ffprobe/packager.'
+            ) from None
+        # If we can't set the permissions for the bundled executables,
+        # we may not be able to run it as a subprocess.
+        if isinstance(ex, PermissionError):
+          raise RuntimeError(
+              'Couldn\'t set the permissions for the bundled '
+              '`shaka-streamer-binaries` executables.\n  Please run as a root.'
```

Comment:
Please remove the "run as root" recommendation.  If it doesn't work at runtime, we need to solve it at packaging time instead, and I'll commit to doing that.  We definitely shouldn't run all of streamer as root, and I don't want to recommend questionable security practices.

If we ultimately can't solve it at packaging time, I'll craft a message here that recommends a specific, targeted command to run as root/admin to fix the permissions.